WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -127,12 +130,26 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
 
             self.viewModel.updateImages(productImageStatuses.images)
         }
+
+        productImageUploader.stopEmittingStatusUpdates(siteID: viewModel.productModel.siteID,
+                                                       productID: viewModel.productModel.productID,
+                                                       isLocalID: !viewModel.productModel.existsRemotely)
     }
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
 
         view.endEditing(true)
+
+        /// Depending on how a VC is presented we need to check different things to know whether it's being dismissed or not.
+        /// A VC presented as the first VC in a navigation controller needs to check if the navigation controller is being dismissed.
+        /// A VC added to an existing navigation controller is dismissed when `isMovingFromParent` is `true`.
+        /// For any other scenario `isBeingDismissed` will do.
+        if isMovingFromParent || isBeingDismissed || navigationController?.isBeingDismissed == true {
```

Comment:
Can we move this to a private computed property? 

Or would it make sense to move this to a `UIViewController` extension for reuse?  🤔

Reply:
good suggestion, moved in https://github.com/woocommerce/woocommerce-ios/pull/7145/commits/48c98d5311a03ffb1280857c56148a3f6087a985. I borrowed this from [WPAuthenticator](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/1adbca81f13756696e566a21d13c51365cc8075f/WordPressAuthenticator/Extensions/UIViewController%2BDismissal.swift#L5-L12) 😄